### ISSUE TITLE
[Bug]: westberks_gov_uk source not showing 2024 collection

### ISSUE THREAD

**Issue by Cha7lie** [mampfes/hacs_waste_collection_schedule#1592]
### I Have A Problem With:

A specific source

### What's Your Problem

**Issue**: The westberks_gov_uk integration / date collections are not updating beyond the end of this year (2023). 

- I have tried testing with the westberks_gov_uk.md example attributes, and have tried UPRN as well as postcode/housenumberorname (as separate configs to rule out an attribute issue).
- Using UPRN attribute results in the error shown in the logs.
- Using postcode/housenumberorname returned recycling for the week of 24-30 Dec 2023 but not rubbish in the first week of Jan 2024.
- I have tried separating out sensors, to have a rubbish sensor and recycling sensor. The issue persists.

### Source (if relevant)

westberks_gov_uk

### Logs

```Shell
2023-12-29 20:22:43.890 ERROR (SyncWorker_2) [waste_collection_schedule.source_shell] fetch failed for source West Berkshire Council:
Traceback (most recent call last):
  File "/config/custom_components/waste_collection_schedule/waste_collection_schedule/source_shell.py", line 134, in fetch
    entries = self._source.fetch()
              ^^^^^^^^^^^^^^^^^^^^
  File "/config/custom_components/waste_collection_schedule/waste_collection_schedule/source/westberks_gov_uk.py", line 101, in fetch
    if not rubbish_data["result"]["nextRubbishDateSubText"]:
           ~~~~~~~~~~~~~~~~~~~~~~^^^^^^^^^^^^^^^^^^^^^^^^^^
KeyError: 'nextRubbishDateSubText'
```

### Relevant Configuration

```YAML
#Waste collection
waste_collection_schedule:
  sources:
    - name: westberks_gov_uk
      args:
        postcode: "my_postcode"
        housenumberorname: "my_house_number"

#Waste collection sensor
sensor:
  - platform: waste_collection_schedule
    name: Waste collection
```

### Checklist Source Error

- [X] Use the example parameters for your source (often available in the documentation) (don't forget to restart Home Assistant after changing the configuration)
- [X] Checked that the website of your service provider is still working
- [X] Tested my attributes on the service provider website (if possible)
- [X] I have tested with the latest version of the integration (master) (for HACS in the 3 dot menu of the integration click on "Redownload" and choose master as version)

### Checklist Sensor Error

- [X] Checked in the Home Assistant Calendar tab if the event names match the types names (if types argument is used)

### Required

- [X] I have searched past (closed AND opened) issues to see if this bug has already been reported, and it hasn't been.
- [X] I understand that people give their precious time for free, and thus I've done my very best to make this problem as easy as possible to investigate.

**Comment by Cha7lie** [comment]
@EPicLURcher - FYI as you're having the same issue with the same source, mentioned in #1590

**Comment by 5ila5** [comment]
fixed this bug in #1597 this fix will be part of version 1.45.0 or is available now as master (git) version (HACS: 3-dot menu -> redownload -> change version to master)

but this bug should solve itself in 2024
